engine/transmittor.py

```diff
@@ -0,0 +1,142 @@
+import json
+from datetime import datetime
+
+class Transmittor:
+    """
+    A class that will transmit the current robot and mission states to the GUI
+    """
+    
+    def __init__(self, robot_state, mission_state):
+        """
+        Store references to states. 
+
+        Parameters:
+            robot_state: an instance encapsulating conditions, measurements, etc. (i.e. all data) for a Robot instance
+            mission_state: an instance encapsulating conditions, measurements, etc. (i.e. all data) for a Mission instance
+        """
+        self.robot_state = robot_state
+        self.mission_state = mission_state
+                 
+    def transmit_to_gui(self):
+        """
+        Call this function to transmit information to the GUI
+        """        
+        self.gather_info()
+    
+    def rmi(self):
```

Comment:
This function has some syntax errors - could you run it in chatgpt and ask it to fix it for you.